### ISSUE TITLE
You can now use BrandBlocker to blacklist optifine? How do you whitelist?

### ISSUE THREAD

**Issue by Hunkulous** [Menacho15/BrandBlocker#1]
I was able to sucessfully blacklist optifine on a PaperMC 1.20.2 server (ViaVersion), they added client branding in the newest 1.20.1 Optifine version branded as 'optifine'.

How do you whitelist optifine/force all players to use optifine? This doesn't kick clients without optifine, although I got the inverse to work.

Do you need to change the 'blocked-brands'?
Sorry if this is a redundant question, i am inexperienced and didn't see an example of a whitelist by anyone on SpigotMC.

Here is my config file:

```
# BrandBlocker by VictorGamer15
#
# Mode types:
# Blacklist: Kick players which have the brand on the brand list below
# Whitelist: Only allow players which have the brand on the brand list below
#
# If you need some help, chat with me on the 'Discussion' tab of the plugin

# Enable the functionality of BrandBlocker
# Users with the permission brandblocker.bypass will not have any restrictions with brands on joining
enable: true

# BrandBlocker mode
mode: whitelist

# List of brands to whitelist or block
blocked-brands:
  - optifine

# Geyser Support
geyser-support: false
geyser-prefix: "*"

# Plugin messages
prefix: "&c&lBrandBlocker &c» "
no-permission: "&7You need the permission &ebrandblocker.usage &7to use this command."
specify-player-name: "&7You need to specify a &eplayer name &7to use this command."
check-succesful: "&e%player% &7entered to the server with the client brand &e%brand%&7."
check-failed: "&7The player &e%player% &7didn't send a client brand packet, or it isn't connected."
config-reload: "&7The main configuration file was &ereloaded&7."
console-log: "%player% was kicked for using %brand%."

kick-message: |-
```

Edit: Sadly the issue here might be actually that the plugin in 1.20.2 seems to output that it kicks a specific user but does not actually kick that user.
I am also using Viaversion but I doubt that is the issue as both the client and server are on 1.20.2

**Comment by Menacho15** [comment]
Hi, it's been more than a year since I've updated BrandBlocker, and I haven't tested it on newer versions. In theory, the whitelist mode should work, because it checks the client brand as good as the blacklist mode, but I have not tested it on newer versions. Maybe I'll rewrite BrandBlocker as the code of it isn't very good.

**Comment by Menacho15** [comment]
Ok, I've tested it, and, apparently, something is broken with the whitelist mode, I'll push an update when possible.

**Comment by Menacho15** [comment]
> Ok, I've tested it, and, apparently, something is broken with the whitelist mode, I'll push an update when possible.

I had BrandBlocker disabled, lol, it's been a long time XD. I tested it on a Paper 1.20.2 server with ViaVersion and ViaBackwards (4.8.1 both versions) and it works without any hassle. I'll update the plugin on SpigotMC. If you still see that players are not being kicked, try using the default kick message, because you apparently have it empty, but I don't know if that can cause issues.

**Comment by Hunkulous** [comment]
Thanks
I think maybe I changed the formatting message on Linux without realizing it would cause issues
It seemed to log the kick message fine

I really appreciate your response

I got the whitelist to work on 1.20.1 without viaversion, but thanks to you; I will go back to 1.20.2 knowing that it works (I must have caused an issue in the confit like you said) which is nice as optifine 1.20.2 is right around the corner about to be released.

the good thing is that BrandBlocker is currently the only publicly available plugin on SpigotMC that can whitelist optifine, or without hardcoded brands.
Thanks to you I have set up a server that requires all my players to use a hardcore darkness resource pack with optifine required.

Great job.
